### PR TITLE
Adding support for plays with Python test fixture 

### PR DESCRIPTION
**Pull request by PhilD71**
### Description
Added support for plays with our new fangled python test fixture
Ported over old ball placement play test + added a new second test 
All new tests can be found here: `ball_placement_play_test.py`

### Testing Done
`ball_placement_play_test.py`

### Resolved Issues
The first of many PR's to resolve (this issue)[https://github.com/UBC-Thunderbots/Software/issues/2558]. Not linking because I don't want it to close this PR

### Length Justification and Key Files to Review

sub 500.
### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

File: src/software/ai/hl/stp/play/ball_placement_play_test.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+from proto.ssl_gc_geometry_pb2 import Vector2
+
+# TODO issue  #2599 - Remove Duration parameter from test
+@pytest.mark.parametrize("run_enemy_ai,test_duration", [(False, 10), (True, 20)])
+def test_two_ai_ball_placement(simulated_test_runner, run_enemy_ai, test_duration):
+
+    # starting point must be Point
+    ball_initial_pos = tbots.Point(2, 2)
+    # placement point must be Vector2 to work with game controller
+    ball_final_pos = Vector2(x=-3, y=-2)
```
Comment (akhilveeraghanta) [inline src/software/ai/hl/stp/play/ball_placement_play_test.py]: lets use our vector  here and convert to the gamecontroller one in the send_ci_input function to make it a bit easier

**Review by akhilveeraghanta — COMMENTED** [review]
LGTM :fire:, nice work!

just got 1 comment

File: src/software/ai/hl/stp/play/play_factory.cpp [hunk 2]
```diff
@@ -9,5 +9,5 @@ std::unique_ptr<Play> createPlay(const TbotsProto::Play& play_proto,
                                  std::shared_ptr<const AiConfig> ai_config)
 {
     return GenericFactory<std::string, Play, AiConfig>::create(
-        TbotsProto::Play_Name(play_proto), ai_config);
+        TbotsProto::Play_PlayName_Name(play_proto.name()), ai_config);
```
Comment (LiCody) [inline src/software/ai/hl/stp/play/play_factory.cpp]:
```suggestion
        TbotsProto::Play::PlayName_Name(play_proto.name()), ai_config);
```
Makes readability slightly better

File: src/software/ai/hl/stp/play/play_factory_test.cpp [hunk 3]
```diff
@@ -13,7 +13,8 @@ class PlayFactoryTest : public testing::Test
 
 TEST_F(PlayFactoryTest, test_shoot_or_pass_play)
 {
-    TbotsProto::Play play_proto = TbotsProto::Play::ShootOrPassPlay;
-    std::unique_ptr<Play> play  = createPlay(play_proto, ai_config);
+    TbotsProto::Play play_proto = TbotsProto::Play();
+    play_proto.set_name(TbotsProto::Play::PlayName::Play_PlayName_ShootOrPassPlay);
```
Comment (LiCody) [inline src/software/ai/hl/stp/play/play_factory_test.cpp]:
```suggestion
    play_proto.set_name(TbotsProto::Play::ShootOrPassPlay);
```
Example here https://developers.google.com/protocol-buffers/docs/cpptutorial#enums-and-nested-classes should work with proto2

File: src/software/ai/hl/stp/play/enemy_ball_placement_play.cpp [hunk 4]
```diff
@@ -81,8 +81,9 @@ void EnemyBallPlacementPlay::ballPlacementWithShadow(
             move_tactics[0]->updateControlParams(
                 placement_point + placement_to_net,
                 placement_to_net.orientation() + Angle::half(), 0);
-            shadow_enemy->updateControlParams(enemy_threats.at(0),
-                                              ROBOT_MAX_RADIUS_METERS * 3);
+            // We need a big shadow distance to avoid "bullying" the robot ball placing.
+            // Otherwise, we get a penalty
+            shadow_enemy->updateControlParams(enemy_threats.at(0), 1.25);
```
Comment (LiCody) [inline src/software/ai/hl/stp/play/enemy_ball_placement_play.cpp]: wheres the 1.25 from? maybe pull it out to a constant

File: src/software/ai/hl/stp/play/ball_placement_play_test.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+    simulated_test_runner.run_test(
+        eventually_validation_sequence_set=eventually_validation_sequence_set,
+        always_validation_sequence_set=always_validation_sequence_set,
+        test_timeout_s=test_duration,
+    )
+
+
+if __name__ == "__main__":
+    sys.exit(pytest.main([__file__, "-svv"]))
```
Comment (LiCody) [inline src/software/ai/hl/stp/play/ball_placement_play_test.py:101]: Can a comment be added for what -svv does

**Review by LiCody — CHANGES_REQUESTED** [review]
LGTM couple nits

File: src/software/ai/hl/stp/play/play_factory.cpp (hunk 2 above)
Comment (PhilD71) [reply src/software/ai/hl/stp/play/play_factory.cpp]: agreed

File: src/software/ai/hl/stp/play/play_factory_test.cpp (hunk 3 above)
Comment (PhilD71) [reply src/software/ai/hl/stp/play/play_factory_test.cpp]: oh cool, I didn't know about that - updating

File: src/software/ai/hl/stp/play/enemy_ball_placement_play.cpp (hunk 4 above)
Comment (PhilD71) [reply src/software/ai/hl/stp/play/enemy_ball_placement_play.cpp]: oops that was supposed to get changed to something else - will change and comment

**Review by jonathanlew — APPROVED** [review]
Nice work!

**Review by LiCody — APPROVED** [review]
LGTM

**Pull request merged**